packages/jest-core/src/TestScheduler.ts

```diff
@@ -176,15 +188,17 @@ export default class TestScheduler {
       showStatus: !runInBand,
     });
 
-    const testRunners: {[key: string]: TestRunner} = Object.create(null);
-    contexts.forEach(({config}) => {
+    const testRunners = Object.create(null);
+    const contextsByTestRunner = new WeakMap<TestRunner, Context>();
+    contexts.forEach(context => {
+      const {config} = context;
       if (!testRunners[config.runner]) {
         const Runner: typeof TestRunner = require(config.runner);
-        testRunners[config.runner] = new Runner(this._globalConfig, {
-          changedFiles: this._context?.changedFiles,
-          sourcesRelatedToTestsInChangedFiles: this._context
-            ?.sourcesRelatedToTestsInChangedFiles,
+        const runner = new Runner(this._globalConfig, {
+          changedFiles: this._context && this._context.changedFiles,
```

Comment:
this looks off - `sourcesRelatedToTestsInChangedFiles` is gone

Reply:
Hmm. Strange got missed. You're right, adding it back